File: P1A5/main (1).py
#ANA ALEJANDRA PARRA ALMAGUER
#00595316
#TI
class Factura: 
    #Atributos incializamos en el constructor __int__
    def __init__(self):
        self.numPieza = ""
        self.descripcion = ""
        self.cantidad = 0
        self.precio = 0.0
        
    def obtenerMontoFactura(self):
        return self.cantidad*self.precio

#FacturaTest
ferre = Factura()
    
ferre.numPieza = input("Ingresa el numero de pieza: ")
ferre.descripcion = input("Descripción: ")
ferre.cantidad = int(input("Cantidad: "))
ferre.precio = float(input("Precio: "))
    
ferre.cantidad = 0 if ferre.cantidad<0 else ferre.cantidad
ferre.precio = 0.0 if ferre.precio<0 else ferre.precio
    
print("Factura: ", ferre.obtenerMontoFactura())
        

File: P1A5/main (1).cs
//ANA ALEJANDRA PARRA ALMAGUER
//00595316
//TI
using System;

public class Factura 
{
    public static void Main(string[] args) 
    {
        Factura ferre = new Factura();

        Console.Write("Ingrese el número de pieza: ");
        ferre.numPieza = Console.ReadLine();

        Console.Write("Agrega la descripción: ");
        ferre.descripcion = Console.ReadLine();

        Console.Write("Cantidad: ");
        ferre.cantidad = int.Parse(Console.ReadLine());

        Console.Write("Precio: ");
        ferre.precio = double.Parse(Console.ReadLine());

        if (ferre.precio < 0) 
        {
            ferre.precio = 0.0;
        }

        if (ferre.cantidad < 0) 
        {
            ferre.cantidad = 0;
        }

        Console.WriteLine("Factura: " + ferre.ObtenerMontoFactura());
    }
    
    // Atributos
    public string numPieza;
    public string descripcion;
    public int cantidad;
    public double precio;

    // Constructor
    public Factura() 
    {
        numPieza = "";
        descripcion = "";
        cantidad = 0;
        precio = 0.0;
    }

    // Método para obtener el monto
    public double ObtenerMontoFactura() 
    {
        return cantidad * precio;
    }
}